src/include/migraphx/matcher.hpp

```diff
@@ -280,6 +280,20 @@ match::matcher_result find_match(module& modl, M&& m)
 MIGRAPHX_DECLARE_ENV_VAR(MIGRAPHX_TRACE_MATCHES)
 
 /// Find matches for an instruction in the module
+//
+// matcher grammar:
+// used_once() means only match an instruction whose output is used only once (otherwise,
+// substituting
+//   it may not improve performance).
+// bind("x") makes an instruction "x" findable by substitution code such as x_ins =
+// r.instructions["x"];
```

Comment:
This is very confusing. `.bind` doesn't make the instruction findable, it just associates a name to the instruction. It returns a matcher as well(so can be further composable). Also, there is no explanation what `r` is in this comment.

This documentation should go with the `bind` method in the `basic_matcher` class(there is also the `bindable_matcher` class but I plan to remove that in the future so dont worry about adding the documentation there).